dashboard/test/models/pd/enrollment_test.rb

```diff
@@ -18,6 +18,94 @@ class Pd::EnrollmentTest < ActiveSupport::TestCase
     assert_equal enrollment, found_enrollment
   end
 
+  # charter + zip  --  success
+  test 'district by type charter and zip, validation succeeds' do
+    enrollment = build :pd_enrollment, school_type: "charter", school_zip: 98144, school_state: nil, school_district_id: nil
+    assert enrollment.valid?
+    assert_equal 0, enrollment.errors.messages.count
```

Comment:
`errors` isn't set until `.valid?` is called (explicitly or implicitly via `save`, etc), so you could drop the 2nd assertion but you do need to check `enrollment.valid?`